online-shop-ui/src/components/products-list/products-list.tsx

```diff
@@ -0,0 +1,65 @@
+import "./products-list.scss";
+import { Product } from "../../data/products";
+import { Link } from "react-router-dom";
+import ProductListItem from "../product-list-item/product-list-item";
+import { useEffect, useState } from "react";
+import { productsService } from "../../services/products.service";
+
+export default function Products() {
+  const [products, setProducts] = useState<Product[] | null>(null);
```

Comment:
I think that it would be nicer to get rid of null and just initialize the products with the empty list.